modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/DataStreamsStatsTransportAction.java

```diff
@@ -158,7 +161,10 @@ protected void shardOperation(
                     maxTimestamp = LongPoint.decodeDimension(maxPackedValue, 0);
                 }
             }
-            return new DataStreamsStatsAction.DataStreamShardStats(indexShard.routingEntry(), storeStats, maxTimestamp);
+            var indexMetadata = indexService.getMetadata();
+            assert indexMetadata != null;
+            double writeLoadForecast = writeLoadForecaster.getForecastedWriteLoad(indexMetadata).orElse(0.0);
```

Comment:
I was not entirely sure if this usage of the `WriteLoadForecaster` is correct, can somebody verify that?

Reply:
The usage is ok except:

- we should preserve `null` rather than mapping it to 0. The difference between "write load is zero" and "write load is missing" is important.

- there's no need to do this for each shard. Instead we should extract the write load from the index metadata in `getResponseFactory`.

Reply:
NB edited^

Reply:
Thanks @DaveCTurner for your feedback, I fixed the things you mentioned, `writeLoadForecast` is now Nullable and extracted only from the write index of the data stream